### PR TITLE
Fixes dialog title announce twice

### LINKED ISSUE #90066: [iOS][A11y] Dialog title announce twice
b/182334959
b/182092337
b/194686096
b/194462962

## Steps to Reproduce
```dart
import 'package:flutter/material.dart';

void main() => runApp(const MyApp());

/// This is the main application widget.
class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  static const String _title = 'Flutter Code Sample';

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: _title,
      home: Scaffold(
        appBar: AppBar(title: const Text(_title)),
        body: const Center(
          child: MyStatelessWidget(),
        ),
      ),
    );
  }
}

/// This is the stateless widget that the main application instantiates.
class MyStatelessWidget extends StatelessWidget {
  const MyStatelessWidget({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return TextButton(
      onPressed: () => showDialog<String>(
        context: context,
        builder: (BuildContext context) => SimpleDialog(
          title: const Text('Select assignment'),
          children: <Widget>[
            SimpleDialogOption(
              onPressed: () { Navigator.pop(context); },
              child: const Text('Treasury department'),
            ),
            SimpleDialogOption(
              onPressed: () { Navigator.pop(context); },
              child: const Text('State department'),
            ),
          ],
        ),
      ),
      child: const Text('Show Dialog'),
    );
  }
}

```

The original issue https://github.com/flutter/flutter/issues/44001 is closed as WAI, but it seems like the original behavior is incorrect for all use cases, and customers have to apply workaround to make it work correctly.

affected dialog

iOS
AlertDialog
SimpleDialog

Android
SimpleDialog

### PR DESCRIPTION
**Pull request by chunhtai**
fixes https://github.com/flutter/flutter/issues/90066

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by chunhtai** [comment]
existing test coverage

File: packages/flutter/test/material/dialog_test.dart [hunk 1]
```diff
@@ -1319,54 +1319,6 @@ void main() {
     );
   });
 
-  testWidgets('AlertDialog widget contains route semantics from title for iOS', (WidgetTester tester) async {
```
Comment (gaaclarke) [inline packages/flutter/test/material/dialog_test.dart:1322]: How come the descriptions says this uses an existing test, but this existing test is being removed?

File: packages/flutter/lib/src/material/dialog.dart [hunk 2]
```diff
@@ -469,6 +469,8 @@ class AlertDialog extends StatelessWidget {
     String? label = semanticLabel;
     switch (theme.platform) {
       case TargetPlatform.iOS:
+        label = '';
```
Comment (gaaclarke) [inline packages/flutter/lib/src/material/dialog.dart:472]: This isn't intuitive why it is the correct thing to do, we should probably add a small comment.

File: packages/flutter/test/material/dialog_test.dart (hunk 1 above)
Comment (chunhtai) [reply packages/flutter/test/material/dialog_test.dart:1322]: This is  testing a wrong behavior

**Review by gaaclarke — APPROVED** [review]
lgtm

**Pull request merged**